src/services/telemetryService.ts

```diff
@@ -102,26 +117,26 @@ export interface UserFlowTelemetry extends TelemetryOptions {
 }
 
 export interface ITelemetryService {
-	sendTelemetryEvent(eventName: string, properties?: TelemetryEventProperties): void;
-	sendRawTelemetryEvent(eventName: string, properties?: RawTelemetryEventProperties): void;
-	sendTelemetryException(error: Error, properties?: TelemetryEventProperties): void;
+	sendTelemetryEvent(gitpodHost: string, eventName: string, properties?: TelemetryEventProperties): void;
+	sendRawTelemetryEvent(gitpodHost: string, eventName: string, properties?: RawTelemetryEventProperties): void;
+	sendTelemetryException(gitpodHost: string, error: Error, properties?: TelemetryEventProperties): void;
 
 	sendUserFlowStatus(status: string, flow: UserFlowTelemetry): void;
 }
 
 export class TelemetryService extends BaseTelemetryReporter implements ITelemetryService {
-	constructor(extensionId: string, extensionVersion: string, key: string, logger: vscode.LogOutputChannel, isProduction: boolean) {
-		const appender = new BaseTelemetryAppender(key, (key) => analyticsClientFactory(key, logger, isProduction));
-		super(extensionId, extensionVersion, appender, {
+	constructor(extensionId: string, extensionVersion: string, segmentKey: string, logger: vscode.LogOutputChannel) {
+		super(extensionId, extensionVersion, {
 			release: os.release(),
 			platform: os.platform(),
 			architecture: os.arch(),
-		});
+		}, gitpodHost => analyticsClientFactory(gitpodHost, segmentKey, logger));
 	}
 
 	sendUserFlowStatus(status: string, flow: UserFlowTelemetry): void {
-		const properties: TelemetryOptions = { ...flow, status };
+		const properties: Partial<TelemetryOptions> = { ...flow, status };
 		delete properties['flow'];
-		this.sendRawTelemetryEvent('vscode_desktop_' + flow.flow, properties);
+		delete properties['gitpodHost'];
```

Comment:
Not sure about this, we still have self hosted installation events and I use this to monitor, how are we gonna differentiate